piControlMain.c

```diff
@@ -296,18 +314,12 @@ static int __init piControlInit(void)
 		// ignore errors
 	}
 
-	if (piDev_g.machine_type == REVPI_CORE) {
-		res = revpi_core_init();
-	} if (piDev_g.machine_type == REVPI_CORE_SE) {
-		res = revpi_core_init();
-	} else if (piDev_g.machine_type == REVPI_CONNECT) {
-		res = revpi_core_init();
-	} else if (piDev_g.machine_type == REVPI_CONNECT_SE) {
-		res = revpi_core_init();
-	} else if (piDev_g.machine_type == REVPI_COMPACT) {
+	if (piDev_g.machine_type == REVPI_COMPACT) {
 		res = revpi_compact_init();
 	} else if (piDev_g.machine_type == REVPI_FLAT) {
 		res = revpi_flat_init();
+	} else if (piDev_g.pibridge_rs485_supported == 1) {
+		res = revpi_core_init();
```

Comment:
In this particular case, I'd leave the enumeration of all products as is. `revpi_core_init()` just *happens* to be called on all PiBridge-capable machines, but calling that function doesn't follow from those machines having a PiBridge. They just happen to share the same initialization code. Making the initialization call hinge on the presence of a PiBridge is confusing.

E.g. if there's a future product which has a PiBridge but needs some additional initialization steps, it might get its own init function which calls down to `revpi_core_init()` and then performs its specific init steps. We'll then have to give that product its own entry in this if-else ladder even though it has a PiBridge.